src/krb5/_set_password.pyx

```diff
@@ -33,14 +42,33 @@ cdef extern from "python_krb5.h":
         krb5_data *result_string
     ) nogil
 
-SetPasswordResult = collections.namedtuple(
-    'SetPasswordResult',
-    [
-        'result_code',
-        'result_code_string',
-        'result_string',
-    ],
-)
+
+
+class SetPasswordResultCode(enum.IntEnum):
+    SUCCESS = KRB5_KPASSWD_SUCCESS
+    MALFORMED = KRB5_KPASSWD_MALFORMED
+    HARDERROR = KRB5_KPASSWD_HARDERROR
+    AUTHERROR = KRB5_KPASSWD_AUTHERROR
+    SOFTERROR = KRB5_KPASSWD_SOFTERROR
+    ACCESSDENIED = KRB5_KPASSWD_ACCESSDENIED
+    BAD_VERSION = KRB5_KPASSWD_BAD_VERSION
+    INITIAL_FLAG_NEEDED = KRB5_KPASSWD_INITIAL_FLAG_NEEDED
+
+    @classmethod
+    def _missing_(cls, value: object) -> typing.Optional[enum.Enum]:
+        if not isinstance(value, int):
+            return None
+        value = int(value)
+
+        new_member = int.__new__(cls, value)
+        new_member._name_ = f"Unknown_NameType_{str(value).replace('-', 'm')}"
```

Comment:
```suggestion
        new_member._name_ = f"Unknown_SetPasswordResultCode_{str(value).replace('-', 'm')}"
```

Reply:
oh sorry, `NameType` resonated like `TypeName` to me:) sure